Assets/Scripts/System/BuilderAssetLoaderSystem.cs

```diff
@@ -0,0 +1,415 @@
+            public int meshes;
+            public int bodies;
+            public int entities;
+        }
+
+        // ReSharper restore CollectionNeverUpdated.Local InconsistentNaming
+
+#pragma warning restore CS0649
+
+        #endregion
+
+        private bool thumbnailRequestCoroutineRunning = false;
+
+        // Dependencies
+        private BuilderAssetLoaderState _loaderState;
+        private EditorEvents _editorEvents;
+        private LoadGltfFromFileSystem _loadGltfFromFileSystem;
+        private IWebRequestSystem _webRequestSystem;
+
+        private BuilderAssetDownLoader assetDownLoader;
+
+        [Inject]
+        public void Construct(BuilderAssetLoaderState loaderState, EditorEvents editorEvents, LoadGltfFromFileSystem loadGltfFromFileSystem, IWebRequestSystem webRequestSystem)
+        {
+            _loaderState = loaderState;
+            _editorEvents = editorEvents;
+            _loadGltfFromFileSystem = loadGltfFromFileSystem;
+            _webRequestSystem = webRequestSystem;
+
+            this.assetDownLoader = new BuilderAssetDownLoader(modelCachePath, webRequestSystem);
+        }
+
+        public void CacheAllAssetMetadata()
+        {
+            // load all asset metadata from the official builder
+            _webRequestSystem.Get("https://builder-api.decentraland.org/v1/assetPacks?owner=default", request =>
+            {
+                var assetData = JsonConvert.DeserializeObject<AssetPacks>(request.webRequest.downloadHandler.text);
+
+                foreach (AssetPacksData assetPack in assetData.data)
+                {
+                    string assetPackPath = _loaderState.assetHierarchy.path + "/" + assetPack.title;
+                    AssetHierarchyItem assetPackHierarchyItem = new AssetHierarchyItem(assetPack.title, assetPackPath);
+
+                    // Builder assets are in categories which are identified by a string name. E.g. "category":"decorations"
+                    Dictionary<String, AssetHierarchyItem> categories = new Dictionary<string, AssetHierarchyItem>();
+
+                    foreach (AssetPacksAsset asset in assetPack.assets)
+                    {
+                        Guid id = Guid.Parse(asset.id);
+
+                        if (!categories.ContainsKey(asset.category))
+                        {
+                            // Capital first letter
+                            string name = char.ToUpper(asset.category[0]) + asset.category.Substring(1);
+                            string path = assetPackPath + "/" + name;
+                            AssetHierarchyItem categoryHierarchyItem = new AssetHierarchyItem(name, path);
+                            categories.Add(asset.category, categoryHierarchyItem);
+                        }
+                        categories[asset.category].assets.Add(new AssetMetadata(asset.name, id, AssetMetadata.AssetType.Model));
+
+
+                        _loaderState.Data.Add(id, new BuilderAssetLoaderState.DataStorage
```

Comment:
This line throws an error when clicking the Refresh button in the asset browser:
```
ArgumentException: An item with the same key has already been added. Key: a2f47727-3f6c-4313-ae76-8034fafa2e5b
System.Collections.Generic.Dictionary`2[TKey,TValue].TryInsert (TKey key, TValue value, System.Collections.Generic.InsertionBehavior behavior) (at <695d1cc93cca45069c528c15c9fdd749>:0)
System.Collections.Generic.Dictionary`2[TKey,TValue].Add (TKey key, TValue value) (at <695d1cc93cca45069c528c15c9fdd749>:0)
Assets.Scripts.System.BuilderAssetLoaderSystem.<CacheAllAssetMetadata>b__11_0 (UnityEngine.Networking.UnityWebRequestAsyncOperation request) (at Assets/Scripts/System/BuilderAssetLoaderSystem.cs:140)
Assets.Scripts.System.WebRequestSystem+<>c__DisplayClass0_0.<Get>g__OnRequestComplete|0 (UnityEngine.AsyncOperation operation) (at Assets/Scripts/System/WebRequestSystem.cs:28)
UnityEngine.AsyncOperation.InvokeCompletionEvent () (at <07c89f7520694139991332d3cf930d48>:0)
```

Reply:
Fixed. The error occurred because the asset cache wasn't properly cleared before being updated.